### ISSUE TITLE
Editor: Cannot use multiple publisher logos

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#5108]
## Bug Description

It seems like the publisher logo section in the Document Panel does not work as expected. Maybe the spec is unclear or we never fully implemented this.

[The spec currently says](https://docs.google.com/document/d/1GJZljicdXcP7oRdywsF3EGUaCVpCHsDrK2gUMubnrbk/edit?pli=1#heading=h.ymf85pwospli):

> Allows the author to pick a publisher logo, as defined in AMP Stories’ metadata guidelines.
> 
> Inherited from global plugin settings, but can be overridden for the story at hand locally.

**[UX / Design specs](https://www.figma.com/file/Cd9iacVknHK62RO4pRwb68/Stories-Editor?node-id=5837%3A4)**

## Expected Behaviour

I should be able to use different publisher logos for different stories.

## Steps to Reproduce

1. Create story 1, select publisher logo A.
1. Publish story 1.
1. See how logo A is selected as the default in the dashboard settings (so far so good)
1. Create story 2, select publisher logo B.
1. Publish story 2.
1. See how logo B is selected as the default in the dashboard settings (so far so good)
1. Edit story 1
1. See logo B in the document panel instead of A (note: the saved story still contains logo A)
1. Save story 1, now the saved markup contains logo B

## Additional Context

- Plugin Version: 1.1.0
- Operating System: macOS
- Browser: Chrome 86;

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by swissspidy** [comment]
@o-fernandez I think this might need your clarification on how things should work

**Comment by o-fernandez** [comment]
Yeah, Sam had a design for this that had a drop-down where you select the publisher logo, and it would show you all the configured publisher logos. It'd also had a link/CTA to the "settings" page for management of the logos. 

This is completely missing from the editor, so will move this to the UX needed column so we can attach the designs and add it to the workspace queue.

**Comment by swissspidy** [comment]
Is there any update on this?

**Comment by swissspidy** [comment]
Another user request: https://wordpress.org/support/topic/select-different-publisher-logo-for-each-story/

@o-fernandez Do you remember where that UX was left? I think even without this we need to somehow fix the issue where changing the publisher logo in the editor changes the default logo for _all_ stories.

**Comment by o-fernandez** [comment]
Hm... I don't. @aaskedall did you get any UX from Sam on selecting a publisher logo from within the story editor? 
Right now we have the ability to upload multiple logos in the settings, but there's only one default and in the story editor when publishing you don't have an easy way to toggle between them. 

> I think even without this we need to somehow fix the issue where changing the publisher logo in the editor changes the default logo for all stories.

Agree, that seems like a bug. @barklund @miina thoughts?

**Comment by miina** [comment]
@o-fernandez @swissspidy 

I think the latest Slack discussion about this topic was [this](https://xwp.slack.com/archives/C01ESBB8ZN1/p1617197455025200).

Short summary:
We kind of [have (old) design for it](https://www.figma.com/file/Cd9iacVknHK62RO4pRwb68/Stories-Editor?node-id=5837%3A4), however, we should be able to easily use the settings page elements from the Dashboard.

There [was an issue](https://github.com/google/web-stories-wp/issues/570) for addressing the publisher logos in the editor that got closed as a dupe but wasn't really a dupe -- we had separate issues for the dashboard and the editor but the editor issue got confused with the dashboard issue.

Conclusion: we should be able to manage with the UX we have and we need to create an issue for it.

**Comment by o-fernandez** [comment]
> Conclusion: we should be able to manage with the UX we have and we need to create an issue for it.

@miina can this be the issue for it? Or is there a need for yet another ticket/issue?

**Comment by miina** [comment]
> @miina can this be the issue for it? Or is there a need for yet another ticket/issue?

💡 Yes, this can be the issue for it 💡

**Comment by bmattb** [comment]
@barklund will reassess if UX needed, or if existing design pattern can be used.

**Comment by spacedmonkey** [comment]
It sounds like publisher logo needs to be stored at the story level and not a global setting. A publisher logo should be like the poster ( featured ) image, with a default value set.

**Comment by swissspidy** [comment]
Removing WP & Infra pod from this until it's clear what our involvement looks like (if needed)

**Comment by barklund** [comment]
Let's try to use the `AdvancedDropdown` component for this and see if it works as intended and looks okay in context.

**Comment by merapi** [comment]
After checking what was [added/changed](https://github.com/google/web-stories-wp/pull/9101) I've decided that it will be better to work a little more on this and utilize the new stuff. (Just resolving conflicts is not an option)

**Comment by bmattb** [comment]
@merapi - Can you confirm the status on this, are you blocked on this at all?

**Comment by csossi** [comment]
Verified in QA
